src/entity/Player.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Player.h"
+#include "../utils/Log.h"
+#include "../Game.h"
+#include "../state/GameState.h"
+#include "../tile/TileDatabase.h"
+
+Player::Player()
+        : Entity(0, std::string("Player"), sf::Vector2f(0,100), sf::FloatRect(0, 32, 16, 32), sf::Texture()) {
+    texture.loadFromFile("res/textures/player.png");
+    sprite = sf::Sprite(texture);
+    sprite.scale(0.5f, -0.5f);
+
+    speed = {0.0f, 0.0f};
+}
+
+void Player::update(std::chrono::microseconds deltaTime) {
+
+    auto div = [](int a, int b)->int{int ret = a/b; return ret>=0? ret: ret-1;};
+
+    const float MOVEMENT_DIV = 2000.0f;
+
+    if(sf::Keyboard::isKeyPressed(sf::Keyboard::W) && touching_ground){
```

Comment:
We should probably use State Machine of some sort to manage state of entities.

Reply:
Right now, one flag like that is fine, but we need to remember to take care of that in the near future.

Reply:
Player won't have another state. There will be a lot more states in NPC. There we will make state machine.